specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1459 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/AzurePrivateCloudResponse"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_ListInResourceGroup": {
+            "$ref": "./examples/PrivateClouds_ListInResourceGroup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VMwareVirtustream/privateClouds/{privateCloudName}": {
+      "get": {
+        "operationId": "PrivateClouds_GetByIdInResourceGroup",
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "get a private cloud in a resource group",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the private cloud."
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/AzurePrivateCloudResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_GetByIdInResourceGroup": {
+            "$ref": "./examples/PrivateClouds_GetByIdInResourceGroup.json"
+          }
+        }
+      },
+      "put": {
+        "operationId": "PrivateClouds_CreateOrUpdate",
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "create a private cloud",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the private cloud."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AzurePrivateCloudRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/AzurePrivateCloudResponse"
+            }
+          },
+          "202": {
```

Comment:
For async operations please use the `x-ms-long-running-operation` extension, see docs [here](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation).  You can also remove the specified `headers` as the various SDK runtimes have this knowledge baked in.

Reply:
This is great! Now I need to figure out the usual way to expose the polling option for the Azure CLI.